keps/core/0002-proxy-workspace.md

```diff
@@ -0,0 +1,134 @@
+# Workspace mounts enhancement proposal
+
+## Summary
+
+To extend spectrum of use cases for kcp, we need to support workspaces that are not
+backed by a kcp cluster itself, but are backed by a mount that can be used to proxy
+request to remote cluster directly or other workspaces. This enables use cases where
+we can start adding external & internal integration into same workspace structure.
+
+## Motivation
+
+This would enable usecases where kcp is used as a proxy to remote clusters, but
+some of the infrastructure is still managed by kcp. For example, kcp could be
+responsible for ArgoCD, Crossplane, or other infrastructure that is more crd-based.
+
+At this point in time communication between kcp and remote clusters is done via
+workspaces abstractions.
+
+In example:
+```
+$ kubectl ws tree
+root:
+root
+  ├── clusters (workspace)
+  │   ├── cluster-proxy (proxy object but shown as a workspace in tree)
+  │   ├── cluster-1 (workspace)
+  ├── infra (workspace)
+  │   ├── argocd (workspace)
+  │   ├── crossplane (workspace)
+  ...
+```
+
+Where `cluster-mount` is a workspace that is backed by a mount that can be used
+to proxy requests to remote clusters. This way we have unified way of interacting
+with remote clusters, but still have ability to manage some of the infrastructure
+via kcp.
+
+### Goals
+
+1. Agree how to make mount functionality as pluggable as possible.
+2. Agree on the scope of the mount functionality.
+3. Agree on the implementation of the mount functionality.
+
+### Non-Goals
+
+Proxy/Mount functionality is not meant to be a replacement for the kcp clusters.
+In addition implementation of the mount functionality itself is out of scope of this
+proposal. We should enable the mount functionality to be implemented by 3rd party
+using kcp as a platform.
+
+## Proposal
+
+Introduce experimental `Mount` API to enable 2 use cases:
+
+1. Mounting remote clusters as workspaces
+2. Softlink - mounting other workspaces as sub-workspaces, where changing the sub-workspace
+   would change the parent workspace as well.
+
+Initially we would put `Mount` into annotation of the workspace, but in the future
+we will promote it to spec of the workspace.
+
+### Mount API
+
+```go
+
+// Mount is a workspace mount that can be used to mount a workspace into another workspace or resource.
+// Mounting itself is done at front proxy level.
+type Mount struct {
+	// MountSpec is the spec of the mount.
+	MountSpec MountSpec `json:"spec,omitempty"`
+	// MountStatus is the status of the mount.
+	MountStatus MountStatus `json:"status,omitempty"`
+}
```

Comment:
am unclear what this struct means. Are you saying we put this into the annotation at the moment, but later put `MountSpec` into the workspace spec and `MountStatus` into the workspace status?

Reply:
Will update